docs/dev/faq.md

```diff
@@ -0,0 +1,113 @@
+
+```python
+adapter.configure_model(
+  "dcim.rackreservation",
+  fields={
+      "units": _define_units,
+  },
+)
+```
+
+The field definition should map the NetBox field to Nautobot and register an importer for the field:
+
+```python
+def _define_units(field: SourceField) -> None:
+  # Map the field from NetBox to Nautobot.
+  # The Nautobot field name is the same as the NetBox field name in this case.
+  field.set_nautobot_field(field.name)
+
+  def importer(source: RecordData, target: DiffSyncBaseModel) -> None:
+      # In NetBox 3.4, units is a `list[int]`; previous versions use a JSON string with a list of strings.
+      units = source.get(field.name, None)
+
+      # Empty values can be ignored.
+      if units in EMPTY_VALUES:
+          return
+
+      # Convert from NetBox format to Nautobot format.
+      if isinstance(units, str):
+          units = json.loads(units)
+          if units:
+              units = [int(unit) for unit in units]
+
+      # Store the value in the target DiffSyncModel instance.
+      setattr(target, field.nautobot.name, units)
+
+  # Register the importer.
+  field.set_importer(importer)
+```
+
+2. Conversion between different field types is supported; however, it can fail depending on the data content. For instance, converting `str` to `int` will fail if the string contains non-numeric characters.
+
+### What has to be done if NetBox or Nautobot changes a relation from 1-to-many to many-to-many?
+
+Relation field conversions are handled by the default importers in most cases, with the following exception:
+
+- NetBox many-to-many relations can't be converted to Nautobot 1-to-many or 1-to-1 relations.
+
+### What has to be done if NetBox or Nautobot adds a new model?
+
+If the model has the same content type and fields as the Nautobot model, then no action is required; the importer will handle it automatically.
+
+Otherwise, it's necessary to define the deviations between the NetBox and Nautobot models in the `nautobot_netbox_importer/diffsync/adapters/<appropriate>.py` file and register the setup function in the `nautobot_netbox_importer/adapter/netbox.py` file using `register_adapter_setup()` as described above.
+
+## What has to be done to support a minor Nautobot version?
+
+No specific changes to tests should be necessary. The only action required is to cover all deviations between NetBox and Nautobot models in the `nautobot_netbox_importer/diffsync/adapters/<appropriate>.py` file as described above.
+
+## What has to be done to support a major NetBox version?
+
+A new major version of the `nautobot-netbox-importer` package should be released to address updated deviations between NetBox and Nautobot models, independent of the previous major version.
+
+## What has to be done to support a major Nautobot version?
+
+Similar to the previous question, a new independent release of the `nautobot-netbox-importer` package should be released.
```

Comment:
ok, so every time a new major version for Nautobot or NetBox is released, a new major version of this app has to be released to match it. I guess this will be documented in a versions matrix or similar

Reply:
Seems to be the best option for me. Optionally it would be possible to support multiple major NetBox releases at once. Depends on the release cadence, and the decision which major release matrix should be supported. This can be reconsidered in the future.

Updated compatibility matrix to cover the current status.